addons/web/static/src/js/fields/relational_fields.js

```diff
@@ -783,8 +783,8 @@ var FieldX2Many = AbstractField.extend({
             return this._super();
         }
         if (this.renderer) {
-            this.renderer.updateState(this.value, {});
             this.currentColInvisibleFields = this._evalColumnInvisibleFields();
+            this.renderer.updateState(this.value, {'columnInvisibleFields': this.currentColInvisibleFields});
             this.pager.updateState({
                 size: this.value.count,
                 columnInvisibleFields: this.currentColInvisibleFields,
```

Comment:
you can also remove this line.  This is actually coming from the original commit, and it is wrong.  The columnInvisibleFields was supposed to be given to the renderer, not the pager